aspnetcore/fundamentals/w3c-logger/index.md

```diff
@@ -0,0 +1,116 @@
+
+### `LoggingFields`
+
+`W3CLoggerOptions.LoggingFields` is an enum flag that configures specific parts of the request and response to log, as well as other information about the connection. `LoggingFields` defaults to include all possible fields except `UserName` and `Cookie`. 
+
+[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=6)]
+
+| Flag | Description | Value |
+| ---- | ----------- | :---: |
+| None | No logging. | 0x0 |
+| `Date` | The date that the activity occurred. | 0x1 |
+| `Time` |  The date that the activity occurred. | 0x2 |
+| `ClientIpAddress` |  The IP address of the client that accessed the server. | 0x4 |
+| `UserName` |  The name of the authenticated user that accessed the server. | 0x8 |
+| `ServerName` |  The name of the server on which the log entry was generated. | 0x10 |
+| `ServerIpAddress` |  The IP address of the server on which the log entry was generated. | 0x20 |
+| `ServerPort` |  The port number the client is connected to. | 0x40 |
+| `Method` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.Method>. | 0x80 |
+| `UriStem` |  Request Path, which includes both the <xref:Microsoft.AspNetCore.Http.HttpRequest.Path> and <xref:Microsoft.AspNetCore.Http.HttpRequest.PathBase>. | 0x100 |
+| `UriQuery` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.QueryString>. | 0x200 |
+| `ProtocolStatus` |  Response <xref:Microsoft.AspNetCore.Http.HttpResponse.StatusCode>. | 0x400 |
+| `TimeTaken` |  The duration of time, in milliseconds, that the action consumed. | 0x800 |
+| `ProtocolVersion` |  Request <xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol>. | 0x1000 |
+| `Host` |  Request <xref:Microsoft.AspNetCore.Http.Headers.HeaderNames.Host>. | 0x2000 |
+| `UserAgent` |  Request <xref:Microsoft.AspNetCore.Http.Headers.HeaderNames.UserAgent>. | 0x4000 |
+| `Cookie` |  Request <xref:Microsoft.AspNetCore.Http.Headers.HeaderNames.Cookie>. | 0x8000 |
+| `Referer` |  Request <xref:Microsoft.AspNetCore.Http.Headers.HeaderNames.Referer>. | 0x10000 |
+| `ConnectionInfoFields` | Flag for logging a collection of properties, about the HTTP Connection, including `ClientIpAddress`, `ServerIpAddress`, and `ServerPort`. | `ClientIpAddress | ServerIpAddress | ServerPort` |
+| `RequestHeaders` | Flag for logging a collection of request headers, including `Host`, `Referer`, and `UserAgent`. | `Host | Referer | UserAgent` |
+| `Request` | Flag for logging a collection of properties about the request, including `UriStem`, `UriQuery`, `ProtocolVersion`, `Method`, and `RequestHeaders`. | `UriStem | UriQuery | ProtocolVersion | Method | RequestHeaders` |
+| `All` | Flag for logging all possible fields. | `Date | Time | ServerName | Method | UriStem | UriQuery | ProtocolStatus | TimeTaken | ProtocolVersion | Host | UserAgent | Referer | ConnectionInfoFields | UserName | Cookie` |
+
+### `FileSizeLimit`
+
+Maximum log file size in bytes. Defaults to 10 MiB.
```

Comment:
I'm not pushing back on this, just curious: why are the descriptions of the options here instead of just having a link to API docs? Do we intend to have both long term or is this a temporary thing?

Reply:
It's copied from the Http Logging doc again - https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-logging/?view=aspnetcore-6.0#loggingfields. Same question as above @Rick-Anderson

Reply:
It's best to not duplicate content. We make exceptions for some APIs. I'm fine with either approach on this. What is `MiB`? Is that an apple thing?

Reply:
MiB is 2^20 whereas MB is 10^6 (1,048,576 vs. 1,000,000)

Reply:
Copy pasting the API docs here doesn't add much value, these fields only make sense to individually document if there will be expanded remarks or examples.

Reply:
@Rick-Anderson should I assign this to you similar to https://github.com/dotnet/AspNetCore.Docs/issues/23421 for once the API docs are up, or should I assign that to myself?

Reply:
> @Rick-Anderson should I assign this to you similar to #23421 for once the API docs are up, or should I assign that to myself?

Assign to me

Reply:
https://github.com/dotnet/AspNetCore.Docs/issues/23422